options/ansi/include/mlibc/file-io.hpp

```diff
@@ -21,7 +23,25 @@ enum class buffer_mode {
 	line_buffer,
 	full_buffer
 };
-
+struct fsetlock : RecursiveFutexLock {
```

Comment:
Rename to `StdioLock` or something like that. Also, it would be good to use composition and not inheritance.